amethyst_renderer/src/sprite/prefab.rs

```diff
@@ -0,0 +1,732 @@
+        Ok(false)
+    }
+}
+
+/// Prefab for loading a full scene with sprites.
+///
+/// When a `render` is encountered during the processing of this prefab, the sheet index
+/// in that will be loaded from the last encountered `sheets`. It is therefore recommended that
+/// all sheets used in the prefab be loaded on the first entity only.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SpriteScenePrefab {
+    /// Sprite sheets
+    pub sheet: Option<SpriteSheetPrefab>,
+    /// Add `SpriteRender` to the `Entity`
+    pub render: Option<SpriteRenderPrefab>,
+    /// Add `Transform` to the `Entity`
+    pub transform: Option<Transform>,
+}
+
+impl<'a> PrefabData<'a> for SpriteScenePrefab {
+    type SystemData = (
+        <SpriteSheetPrefab as PrefabData<'a>>::SystemData,
+        <SpriteRenderPrefab as PrefabData<'a>>::SystemData,
+        <Transform as PrefabData<'a>>::SystemData,
+    );
+    type Result = ();
+
+    fn add_to_entity(
+        &self,
+        entity: Entity,
+        system_data: &mut Self::SystemData,
+        entities: &[Entity],
+    ) -> Result<(), Error> {
+        if let Some(render) = &self.render {
+            render.add_to_entity(entity, &mut system_data.1, entities)?;
+        }
+        if let Some(transform) = &self.transform {
+            transform.add_to_entity(entity, &mut system_data.2, entities)?;
+        }
+        Ok(())
+    }
+
+    fn load_sub_assets(
+        &mut self,
+        progress: &mut ProgressCounter,
+        system_data: &mut Self::SystemData,
+    ) -> Result<bool, Error> {
+        let mut ret = false;
+        if let Some(ref mut sheet) = &mut self.sheet {
+            if sheet.load_sub_assets(progress, &mut system_data.0)? {
+                ret = true;
+            }
+            let sheet = match sheet {
+                SpriteSheetPrefab::Handle(handle) => handle.clone(),
+                _ => unreachable!(),
+            };
+            ((system_data.1).1).0.push(sheet);
+        }
+        if let Some(ref mut render) = &mut self.render {
+            render.load_sub_assets(progress, &mut system_data.1)?;
+        }
+        Ok(ret)
+    }
+}
+
+impl Sprites {
+    fn build_sprites(&self) -> Vec<Sprite> {
+        match self {
+            Sprites::List(list) => list.build_sprites(),
+            Sprites::Grid(grid) => grid.build_sprites(),
+        }
+    }
+}
+
+impl SpriteList {
+    pub(crate) fn build_sprites(&self) -> Vec<Sprite> {
+        self.sprites
+            .iter()
+            .map(|pos| {
+                Sprite::from_pixel_values(
+                    self.width,
+                    self.height,
+                    pos.width,
+                    pos.height,
+                    pos.x,
+                    pos.y,
+                    pos.offsets.unwrap_or([0.0; 2]),
+                )
+            })
+            .collect()
+    }
+}
+
+impl SpriteGrid {
+    fn width(&self) -> u32 {
```

Comment:
>  think we should rename the field to `texture_width`

Should this be renamed for internal use only?
Right now I use the `#[serde(rename = "spritesheet_width")]` for compatibility with the `SpriteSheetFormat`.
In my opinion, I have to rename these fields to `texture_width / texture_height` everywhere, because they better describe requirements.
But it will change the current format.

Reply:
ah, ya renaming it everywhere is a better idea.
I don't think it hurts to change the current format; other way round, it would help reduce mistakes -- intuitively I would have filled in the subset of the texture's width/height. It's not yet released, so there wouldn't be migration errors too.